mindmeld/text_preparation/text_preparation_pipeline.py

```diff
@@ -0,0 +1,530 @@
+        """
+        normalized_tokens = self.get_normalized_tokens_from_text(text)
+        normalized_text = " ".join([t["entity"] for t in normalized_tokens])
+        return normalized_text
+
+    def get_normalized_tokens_from_text(self, text):
+        """
+        Args:
+            text (str): Text to normalize.
+        Returns:
+            normalized_tokens (List[Dict]): Normalized tokens represented as dictionaries.
+                For Example:
+                    norm_token = {
+                        "entity": "order",
+                        "raw_entity": "order",
+                        "raw_token_index": 1,
+                        "raw_start": 1
+                    }
+        """
+        raw_tokens = self.tokenizer.tokenize(text)
+        return self._normalize_tokens(raw_tokens)
+
+    def _normalize_tokens(self, raw_tokens):
+        """Normalize individual token dicts produced by Tokenizers.
+        Args:
+            raw_tokens (List(Dict)): List of raw tokens represented as dictionaries.
+        Returns:
+            normalized_tokens (List[Dict]): Normalized tokens represented as dictionaries.
+                For Example:
+                    norm_token = {
+                        "entity": "order",
+                        "raw_entity": "order",
+                        "raw_token_index": 1,
+                        "raw_start": 1
+                    }`
+        """
+        normalized_tokens = []
+        for i, raw_token in enumerate(raw_tokens):
+            if not raw_token["text"]:
+                continue
+            normalized_texts = self._normalize_single_token(raw_token["text"])
+            if len(normalized_texts) > 0:
+                for token_text in normalized_texts:
+                    normalized_tokens.append(
+                        {
+                            "entity": token_text,
+                            "raw_entity": raw_token["text"],
+                            "raw_token_index": i,
+                            "raw_start": raw_token["start"],
+                        }
+                    )
+        return normalized_tokens
+
+    def _normalize_single_token(self, text):
+        """Normalize an individual token by processing text with all normalizers.
+
+        Args:
+            text (str): Text to normalize.
+        Returns:
+            normalized_texts (List[str]): Normalized texts. We tokenize the post-norm text and
+                split the entity if possible.
+        """
+        normalized_text = text
+        for normalizer in self.normalizers:
+            normalized_text = TextPreparationPipeline.modify_around_annotations(
+                text=normalized_text, function=normalizer.normalize
+            )
+        # We tokenize the post-norm text and split the entity if possible
+        # Ex: normalize("o'clock") -> "o clock" -> ["o", "clock"]
+        return [t["text"] for t in self.tokenize(normalized_text)]
```

Comment:
why do we do this? It seems like it's the caller's job to do tokenization. In general, I would like to keep these functions simple. So `"o'clock"` -> `o clock` here and then the caller can do whatever with it.